src/app/auth/handleRequest.ts

```diff
@@ -64,4 +68,31 @@ async function loggedinUserName(bsn: string, apiClient: ApiClient) {
     console.error('Error getting username');
     return 'Onbekende gebruiker';
   }
+}
+
+/**
+ * Extract bsn from token claims
+ *
+ * The bsn can be found in several fields: either the 'sub' claim
+ * or a yivi-specific field can be used. This function tries (in order)
+ * to get a valid BSN from:
+ * - sub
+ * - pbdf.gemeente.bsn.bsn
+ * - irma-demo.gemeente.personalData.bsn
+ *
+ * @param claims an IdTokenClaims object
+ * @returns {BSN|false}
+ */
+export function bsnFromClaims(claims: IdTokenClaims): Bsn|false {
+  const possibleClaims = ['sub', 'pbdf.gemeente.bsn.bsn', 'irma-demo.gemeente.personalData.bsn'];
```

Comment:
Wat zorgt er voor dat ik in productie niet met demo attributen in kan loggen nu? Alleen de OIDC claim die wordt meegestuurd?

Reply:
En het feit dat de hele Yivi-login nog gefeature-flagged is, maar als je zelf de url opgeeft kan dat natuurlijk wel. En ook met die demo. Wat niet ok is als je daar een 'echt' bsn in weet te krijgen. Ga ik aanpassen, denk dat ook het checken wat de bron van de claim is geen kwaad kan, zodat ook alleen Irma-attributen kunnen als je een Yivi-login doet. Ga ik eerst toevoegen. Goeie!

Reply:
Daar een echt BSN in krijgen is niet ingewikkeld omdat je voor het Yivi demo schema zelf attributen in mag vullen.